packages/grid/x-grid/src/XGrid.test.tsx

```diff
@@ -220,6 +220,47 @@ describe('<XGrid />', () => {
       fireEvent.click(header);
       expect(getColumnValues()).to.deep.equal(['Puma', 'Nike', 'Adidas']);
     });
+
+    it('should reset the sortedRows state when rows prop change', async () => {
```

Comment:
The bug can be reproduced on the DataGrid: https://codesandbox.io/s/fancy-brook-hf3fw?file=/src/App.js.
So I think that the test should be in `DataGrid.test.tsx`.

Reply:
I think that tests should be as end-to-ends as possible.

```suggestion
    it('should keep rows sorted when rows prop change', async () => {
```